src/java/org/apache/cassandra/cql3/selection/VectorSelector.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.selection;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.google.common.base.Preconditions;
+
+import org.apache.cassandra.cql3.Lists;
+import org.apache.cassandra.cql3.QueryOptions;
+import org.apache.cassandra.db.filter.ColumnFilter;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.VectorType;
+import org.apache.cassandra.exceptions.InvalidRequestException;
+import org.apache.cassandra.io.util.DataInputPlus;
+import org.apache.cassandra.io.util.DataOutputPlus;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.transport.ProtocolVersion;
+
+public class VectorSelector extends Selector
+{
+    protected static final SelectorDeserializer deserializer = new SelectorDeserializer()
+    {
+        protected Selector deserialize(DataInputPlus in, int version, TableMetadata metadata) throws IOException
+        {
+            VectorType<?> type = (VectorType<?>) readType(metadata, in);
+            List<Selector> elements = new ArrayList<>(type.dimension);
+            for (int i = 0; i < type.dimension; i++)
+                elements.add(serializer.deserialize(in, version, metadata));
+
+            return new VectorSelector(type, elements);
+        }
+    };
+
+    /**
+     * The vector type.
+     */
+    private final VectorType<?> type;
+
+    /**
+     * The list elements
+     */
+    private final List<Selector> elements;
+
+    private VectorSelector(VectorType<?> type, List<Selector> elements)
+    {
+        super(Kind.VECTOR_SELECTOR);
+        Preconditions.checkArgument(elements.size() == type.dimension, "Unable to create a vector select of type %s from %s elements", type.asCQL3Type(), elements.size());
```

Comment:
Nit: break the long line, for example:
```suggestion
        Preconditions.checkArgument(elements.size() == type.dimension,
                                    "Unable to create a vector select of type %s from %s elements",
                                    type.asCQL3Type(),
                                    elements.size());
```